gdbstub_arch/src/arm/reg/id.rs

```diff
@@ -31,6 +33,6 @@ impl RegId for ArmCoreRegId {
             25 => Self::Cpsr,
             _ => return None,
         };
-        Some((reg, 4))
+        Some((reg, Some(NonZeroUsize::new(4).unwrap())))
```

Comment:
Any recommendations for these `unwrap()` calls? I know they're not allowed in `gdbstub` code, but this will just get optimized out by the compiler.

Reply:
I mean, since you're already in a function that returns `Option<_>`, you can just use `NonZeroUsize::new(x)?`.

At the end of the day, any reasonably optimizing compiler should generate equivalent code when using `?` or `unwrap` in these trivial cases (i.e: it'll elide the None branch entirely)

Reply:
👍 using the `?` operator is a nicer shorthand :)
I wish there was a `new()` compile-time constructor that threw a compiler error if the constant evaluated to 0 - that'd be a bit nicer.

Reply:
_sigh,_ preaching to the choir...

Reply:
😂 time to submit some RFCs to Rust I guess